[stable/external-dns] Skip mounting when using managed identity extension

austinorth opened the pull request:
#### What this PR does / why we need it:
Adds a conditional which skips mounting `/etc/kubernetes` resources if `azure.useManagedIdentityExtension=true` and bumped chart version. From #18494:
>When running ExternalDNS in Azure together with aad-pod-identity and the deployment option useManagedIdentityExtension=true it's not necessary to mount a hostPath to the pod to get credentials to manage the Azure DNS zone.

Signed-off-by: Austin Orth <aorth@niche.com>


#### Which issue this PR fixes
  - fixes #18494

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

CC: @bitnami-bot

Commit:
[stable/external-dns] Managed identity mount tweak

Added a conditional which skips mounting resources if
azure.useManagedIdentityExtension=true and bumped chart version.

Signed-off-by: Austin Orth <aorth@niche.com>

Commit:
[stable/external-dns] Corrected version bump

Accidentally bumped the app version instead of the chart version in
previous commit. Fixed now.

Signed-off-by: Austin Orth <aorth@niche.com>

juan131 commented on stable/external-dns/templates/deployment.yaml:
```diff
@@ -302,6 +302,7 @@ spec:
           {{- if or .Values.azure.secretName (and .Values.azure.resourceGroup .Values.azure.tenantId .Values.azure.subscriptionId) }}
           mountPath: /etc/kubernetes/
+          {{- else if .Values.azure.useManagedIdentityExtension }}
           {{- else }}
```
What about the following approach?
```diff
-           {{- else if .Values.azure.useManagedIdentityExtension }}
-           {{- else }}
+ .         {{- else if not .Values.azure.useManagedIdentityExtension }}
```

austinorth replied:
That reads a little clearer! I'll make a new commit.

juan131 commented on stable/external-dns/Chart.yaml:
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: external-dns
-version: 2.10.0
+version: 2.11.0
```
Please bump patch version instead:
```diff
- version: 2.11.0
+ version: 2.10.1
```

austinorth replied:
Got it, will update this shortly!

juan131 reviewed (changes requested):
Hi @austinorth 

Thanks so much for your PR!! Could you take a look to my suggestions?

Commit:
[stable/external-dns] Implemented suggestions

Changed conditional to use `if not` as suggested and changed the chart
version to patch level instead of minor.

Signed-off-by: Austin Orth <aorth@niche.com>

juan131 reviewed (approved):
/ok-to-test
/lgtm
